### PR TITLE
chore(ci): add Go scanning and update dependencies

### PR DESCRIPTION
**Pull request by rahul2393**
Close https://github.com/aquasecurity/trivy/issues/997

### CODE REVIEW

**Comment by rahul2393** [comment]
Screenshot 2021-05-18 at 8 18 51 PM

**Comment by rahul2393** [comment]
Screenshot 2021-05-18 at 8 31 03 PM

File: go.mod [hunk 1]
```diff
@@ -53,3 +53,6 @@ require (
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
 	k8s.io/utils v0.0.0-20201110183641-67b214c5f920
 )
+replace github.com/containerd/containerd => github.com/containerd/containerd v1.4.4
+replace golang.org/x/crypto => golang.org/x/crypto v0.0.0-20201216223049-8b5274cf687f
```
Comment (knqyf263) [inline go.mod:57]: Do not use `replace`, but update dependencies.

**Review by knqyf263 — CHANGES_REQUESTED** [review]

File: .github/workflows/scan.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Scan
```
Comment (knqyf263) [inline .github/workflows/scan.yaml:1]:
```suggestion
name: Scan Vulnerabilities
```

**Review by knqyf263 — COMMENTED** [review]
Did you run `go mod tidy`?

File: .github/workflows/scan.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+on: [push, pull_request]
+jobs:
+  build:
+    name: Scan repo
```
Comment (knqyf263) [inline .github/workflows/scan.yaml:5]:
```suggestion
    name: Scan Go vulnerabilities
```

**Comment by rahul2393** [comment]
> Did you run `go mod tidy`?

yes

**Review by knqyf263 — APPROVED** [review]

**Comment by piyush94** [comment]
What is meant by "add Go scanning" in the PR title? Did Trivy was not capable to scan Go packages/dependencies earlier?

**Comment by knqyf263** [comment]
It was added in v0.17.0.
https://github.com/aquasecurity/trivy/discussions/966

**Pull request merged**